File: homework20180917/app/src/main/java/com/example/kehtolaulu/homework_2018_09_17/MainActivity.java
package com.example.kehtolaulu.homework_2018_09_17;

import android.app.Activity;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_ID = 1;
    private static final int REQUEST = 0;

    TextView tvEmail;
    TextView tvName;
    TextView tvPhoneNumber;
    Button btnEdit;
    Button btnSend;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvEmail = findViewById(R.id.tv_email);
        tvName = findViewById(R.id.tv_name);
        tvPhoneNumber = findViewById(R.id.tv_phone_number);
        btnEdit = findViewById(R.id.btn_edit);
        btnSend = findViewById(R.id.btn_send);

        btnEdit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
                startActivityForResult(intent, REQUEST);
            }
        });


        btnSend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String send = tvName.getText().toString();
                Intent intentSend = new Intent(Intent.ACTION_SEND);
                intentSend.putExtra(Intent.EXTRA_TEXT, send);
                intentSend.setType("text/plain");
                startActivityForResult(intentSend, REQUEST_ID);
            }
        });
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (data != null) {
            if (resultCode == Activity.RESULT_CANCELED) {
                Toast.makeText(this, "You have cancelled", Toast.LENGTH_LONG).show();
            }
            String filledEmail = data.getStringExtra("email");
            String filledName = data.getStringExtra("name");
            String filledNumber = data.getStringExtra("number");

            tvEmail.setText(filledEmail);
            tvName.setText(filledName);
            tvPhoneNumber.setText(filledNumber);
        }
    }
}

File: homework06/app/src/main/java/com/example/kehtolaulu/homework06/MainActivity.kt
package com.example.kehtolaulu.homework06

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.os.CountDownTimer
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var etTimeMin: EditText
    private lateinit var etTimeSec: EditText
    private lateinit var btnSet: Button
    private lateinit var btnCancel: Button
    private lateinit var tvLeft: TextView
    private lateinit var alarmManager: AlarmManager
    private lateinit var pendingIntent: PendingIntent
    private lateinit var timer: CountDownTimer

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        etTimeMin = findViewById(R.id.et_min)
        etTimeSec = findViewById(R.id.et_sec)
        tvLeft = findViewById(R.id.tv_left)
        btnSet = findViewById(R.id.btn_set)
        btnCancel = findViewById(R.id.btn_cancel)

        btnCancel.setOnClickListener {
            cancelTimer()
            tvLeft.text = ""
        }

        btnSet.setOnClickListener { startTimer() }
    }

    private fun cancelTimer() {
        alarmManager.cancel(pendingIntent)
        Toast.makeText(this, "You've cancelled", Toast.LENGTH_LONG).show()
        timer.cancel()
    }

    private fun startTimer() {
        val timeOnTimer = (Integer.parseInt(etTimeMin.text.toString()) * 60000 + Integer.parseInt(etTimeSec.text.toString()) * 1000).toLong()
        startTimer(timeOnTimer)
        alarmManager = getSystemService(Context.ALARM_SERVICE) as AlarmManager

        val intentTimer = Intent(this, AlertReceiver::class.java)

        pendingIntent = PendingIntent.getBroadcast(this, 555, intentTimer, PendingIntent.FLAG_CANCEL_CURRENT)
        alarmManager.set(AlarmManager.RTC_WAKEUP, System.currentTimeMillis() + timeOnTimer, pendingIntent)
    }

    private fun startTimer(timeOnTimer: Long) {
        timer = object : CountDownTimer(timeOnTimer, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                tvLeft.text = "" + millisUntilFinished / 60000 + " minutes " + millisUntilFinished % 60000 / 1000 + " seconds left"
            }

            override fun onFinish() {
                Toast.makeText(applicationContext, "Time is out!", Toast.LENGTH_LONG)
                        .show()
            }
        }.start()
    }
}